gatekeeper: add dynamic log type for gatekeeper

cjdoucette opened the pull request:
For Gatekeeper activity that is not related to individual blocks
(such as libraries), this patch adds a dynamic logging type and
removes any lingering references in the Gatekeeper code to
built-in static library log types.

AltraMayor commented on lib/net.c:
```diff
@@ -1333,17 +1331,29 @@ gatekeeper_init_network(struct net_config *net_conf)
 		return -1;
 
+	gatekeeper_logtype = rte_log_register("gatekeeper");
+	if (gatekeeper_logtype < 0) {
```
Registering the `gatekeeper` log type here is too late. `config_gatekeeper()`, for example, may log something before reaching this line. My suggestion is to register the `gatekeeper` log type just after Gatekeeper gets the options from the command line. Thanks to the latest merged pull request, it's now easy to add options to the command line. Please add an option for the log level that the `gatekeeper` log type should be initialized, and another option for the global log level.

AltraMayor replied:
You are right, reusing EAL here is better. Please document those parameters on the wiki because that is not an obvious solution.

AltraMayor commented on include/gatekeeper_net.h:
```diff
@@ -30,6 +30,16 @@
+
+#define G_LOG(level, ...)		\
+	rte_log(RTE_LOG_ ## level,	\
+	gatekeeper_logtype, "GATEKEEPER: " __VA_ARGS__)
```
After the change of initialization of `gatekeeper_logtype` (see another comment), these definitions here should go to `include/gatekeeper_main.h`. Review the `#include "gatekeeper_net.h"` that this patch was adding.

cjdoucette commented:
I rebased and registered `"gatekeeper"` as a log type before parsing the EAL options so that it can be set from the command line. I also moved the definition of `G_LOG()` to main/main.c, but kept the log level and type state in `struct net_config` so that it can be altered using the Dynamic Config.

I also kept it all as one patch because I think your idea to split it was based on comments from before, but I can split it as needed.

cjdoucette commented:
I'm a little bit hesitant about the current setup because the log levels in the configuration will overwrite the log levels specified in the command line after the initial setup, so the command line options are only for early logging. If we go forward with this I'll make a note of that on the wiki.

cjdoucette commented:
Ready for another review.

AltraMayor commented on lib/net.c:
```diff
@@ -1330,6 +1330,8 @@ gatekeeper_init_network(struct net_config *net_conf)
 		return -1;
 
+	net_conf->log_type = gatekeeper_logtype;
+
```
Shouldn't we be setting the log level of the Gatekeeper type to `net_conf->log_level` here?

cjdoucette replied:
Yes, thanks. The latest rebase has this in it.
